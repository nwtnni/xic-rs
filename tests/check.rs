use std::fmt;
use std::path::Path;

struct Snapshot(Result<(), xic::Error>);

impl fmt::Display for Snapshot {
    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
        match &self.0 {
            Ok(()) => write!(fmt, "Valid Xi Program"),
            Err(error) => write!(fmt, "{}", error),
        }
    }
}

#[test_generator::test_resources("tests/check/*.xi")]
pub fn check(path: &str) {
    let lexed = xic::api::lex(Path::new(path)).unwrap();
    let parsed = xic::api::parse(lexed).unwrap();
    let checked = xic::api::check(Path::new("tests/check"), &parsed).map(|_| ());

    insta::assert_display_snapshot!(path, Snapshot(checked));
}
