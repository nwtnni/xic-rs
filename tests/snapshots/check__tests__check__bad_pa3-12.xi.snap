---
source: tests/check.rs
assertion_line: 25
expression: Snapshot(checked)
---
2:9 error:Cannot initialize with a procedure
